homeassistant/components/surepetcare/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for Sure Petcare integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from surepy import Surepy
+from surepy.exceptions import SurePetcareAuthenticationError, SurePetcareError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_TOKEN, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN, SURE_API_TIMEOUT
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    try:
+        surepy = Surepy(
+            data[CONF_USERNAME],
+            data[CONF_PASSWORD],
+            auth_token=None,
+            api_timeout=SURE_API_TIMEOUT,
+            session=async_get_clientsession(hass),
+        )
+
+        token = await surepy.sac.get_token()
+
+    except SurePetcareAuthenticationError as exp:
+        raise InvalidAuth from exp
+
+    except SurePetcareError as exp:
+        raise CannotConnect from exp
+
+    return {CONF_TOKEN: token}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Sure Petcare."""
+
+    VERSION = 1
+
+    async def async_step_import(self, import_info):
+        """Set the config entry up from yaml."""
+        return await self.async_step_user(import_info)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
```

Comment:
```suggestion
        except SurePetcareAuthenticationError:
```

Reply:
Since the library already has named exceptions, we can avoid creating new ones and let the `validate_input` function raise